### DISCUSSION TITLE
Sound Detection Trigger

### DISCUSSION THREAD

**Discussion by shinra358** [ikemen-engine/Ikemen-GO#1977] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Yes, currently there's no way to detect if a particular sound in playing in the same state as the sound call which is a nightmare for helpers detecting the time and state the root char is in and playing the sound from the helper. And the sound ends up constantly overlapping and replaying when you dont want it to.

### Describe the solution you'd like

if you can a sound for ex. S10,0

we can use a trigger 'IsSound(S10,0)' to say not when that particular sound is playing, should you only play the sound like so:
 !IsSound(S10,0)

**Comment by Kiwi1444** [comment]
Agreed. I'm already offered similiar idea: https://github.com/ikemen-engine/Ikemen-GO/discussions/1902

**Comment by K4thos** [comment, marked as answer]
implemented in #2273
